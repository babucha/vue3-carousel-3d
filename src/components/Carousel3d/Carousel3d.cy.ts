import Carousel3d from './Carousel3d.vue'
import Slide from '../Slide/Slide.vue'

import { h } from 'vue'

describe('<Carousel3d />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-vue
    cy.mount(Carousel3d, {
      slots: {
        default: () => [
          h(Slide, {}, h('img', { src: 'https://via.placeholder.com/360x270' }, {})),
          h(Slide, {}, h('img', { src: 'https://via.placeholder.com/360x270' }, {})),
          h(Slide, {}, h('img', { src: 'https://via.placeholder.com/360x270' }, {}))
        ]
      }
    })
  })
})
