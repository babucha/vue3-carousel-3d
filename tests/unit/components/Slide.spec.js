const utils = require('../utils')

import Carousel3d from '../../../src/components/Carousel3d/Carousel3d.vue';
import Slide from "../../../src/components/Slide/Slide.vue";

import { mount, flushPromises } from "@vue/test-utils";
import {createApp, h} from 'vue'

describe('Slide', () => {
    test('should mount successfully', async () => {
        const slide = mount(Slide)
        const wrapper = mount(Carousel3d, {
            slots: {
                default: [Slide]
            }
        })

        await flushPromises()
        await wrapper.vm.$nextTick()
        await wrapper.vm.$nextTick()
        // console.log(slide.html())
        console.log(wrapper.html())

        const slideInstance = wrapper.vm.$slots.default()[0]
        console.log(slideInstance.type)
        expect(slideInstance).toBeDefined()

        return utils.expectToMatchSnapshot(wrapper.vm)

        // const vm = createApp({
        //     render() {
        //         return h(Carousel3d, {}, () => [h(Slide)]);
        //     },
        // })
        // .mount(document.createElement('div'));

        // await flushPromises()

        // console.log(vm)
        // const carouselInstance = vm.$children[0]
        // const slideInstance = vm.$slots.default()[0]
        // expect(slideInstance._isMounted).toBe(true)

        // return utils.expectToMatchSnapshot(vm)
    })

//     it('should unmount successfully', () => {
//         const vm = new Vue({
//             el: document.createElement('div'),
//             render: (h) => h(Carousel3d, {}, [h(Slide)])
//         })
//         const carouselInstance = vm.$children[0]
//         const slideInstance = carouselInstance.$children[0]
//         slideInstance.$destroy()
//         expect(slideInstance._isDestroyed).toBe(true)

//         return utils.expectToMatchSnapshot(vm)
//     })

//     it('should return style object when 3 slides are added to the slots', () => {
//         const vm = new Vue({
//             el: document.createElement('div'),
//             render: (h) => h(Carousel3d, {}, [h(Slide), h(Slide), h(Slide)])
//         })
//         const carouselInstance = vm.$children[0]
//         const slideInstance = carouselInstance.$children[0]
//         const o = slideInstance.calculatePosition(0, true)

//         expect(o.transform).toMatch(/translateX/)
//         expect(o.top).toBe(0)

//         return utils.expectToMatchSnapshot(vm)
//     })

//     it('should render a scoped slot', () => {
//         const vm = new Vue({
//             el: document.createElement('div'),
//             render: (h) => h(Carousel3d, {}, Array.apply(null, { length: 5 }).map((k, index) => h(Slide, {
//                 /* adds the index prop */
//                 props: {
//                     index: index
//                 },
//                 /* add a default scoped slot */
//                 scopedSlots: {
//                     default: props => h('div', {
//                         attrs: {
//                             /* with left and right data attrs */
//                             'data-left-index': props.leftIndex,
//                             'data-right-index': props.rightIndex,
//                             'data-index': props.index,
//                             'data-is-current': props.isCurrent
//                         }
//                     }, index)
//                 }
//             })))
//         })

//         return utils.expectToMatchSnapshot(vm)
//     })
})
