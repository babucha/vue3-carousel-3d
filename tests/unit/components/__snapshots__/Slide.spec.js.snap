// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slide > should mount successfully 1`] = `
<div class="carousel-3d-container" style="height: 272px;">
  <div class="carousel-3d-slider" style="width: 362px; height: 272px;">
    <div class="carousel-3d-slide" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;"></div>
  </div>
</div>
`;

exports[`Slide > should render a scoped slot 1`] = `
<div class="carousel-3d-container" style="height: 272px;">
  <div class="carousel-3d-slider" style="width: 362px; height: 272px;">
    <div class="carousel-3d-slide current" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;">
      <div data-index="0" data-is-current="true" data-left-index="-1" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide right-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(240px) translateZ(-400px) rotateY(-35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="1" data-left-index="-1" data-right-index="0"></div>
    </div>
    <div class="carousel-3d-slide right-2" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(480px) translateZ(-500px) rotateY(-35deg); top: 0px; z-index: 997; opacity: 1; visibility: visible;">
      <div data-index="2" data-left-index="-1" data-right-index="1"></div>
    </div>
    <div class="carousel-3d-slide left-2" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-480px) translateZ(-500px) rotateY(35deg); top: 0px; z-index: 997; opacity: 1; visibility: visible;">
      <div data-index="3" data-left-index="1" data-right-index="-1"></div>
    </div>
    <div class="carousel-3d-slide left-1" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms; transform: translateX(-240px) translateZ(-400px) rotateY(35deg); top: 0px; z-index: 998; opacity: 1; visibility: visible;">
      <div data-index="4" data-left-index="0" data-right-index="-1"></div>
    </div>
  </div>
</div>
`;

exports[`Slide > should return style object when 3 slides are added to the slots 1`] = `
<div class="carousel-3d-container" style="height: 272px;">
  <div class="carousel-3d-slider" style="width: 362px; height: 272px;">
    <div class="carousel-3d-slide" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;"></div>
    <div class="carousel-3d-slide" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;"></div>
    <div class="carousel-3d-slide" style="border-width: 1px; width: 362px; height: 272px; transition:  transform 500ms,                opacity 500ms,                visibility 500ms;"></div>
  </div>
</div>
`;
